src/Microsoft.AspNetCore.Razor.OmniSharpPlugin/DefaultProjectChangePublisher.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.IO;
+using Microsoft.AspNetCore.Razor.OmniSharpPlugin.StrongNamed.Serialization;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Microsoft.AspNetCore.Razor.OmniSharpPlugin
+{
+    [Shared]
+    [Export(typeof(ProjectChangePublisher))]
+    [Export(typeof(OmniSharpProjectSnapshotManagerChangeTrigger))]
+    internal class DefaultProjectChangePublisher : ProjectChangePublisher, OmniSharpProjectSnapshotManagerChangeTrigger
```

Comment:
This is the piece that ties everything together in the plugin. It listens to change events from the snapshot manager and then serializes the projects down to the file system. Nice thing about this is that the project snapshot manager handles ensuring everything is synchronized correctly, doesn't re-dispatch on no-op changes, and overall just manages every bit of state we could ever want.